### DISCUSSION TITLE
VS Code | Does this extension use any analytics or report the components source code?

### DISCUSSION THREAD

**Discussion by lgibso34** [fwouts/previewjs#479] [Q&A] [answered]
I'd like to try this at my company but I work for a bank and know there are many security considerations when installing any new applications or possibly VS Code extensions.

Does this extension report any analytics or send out source code?

Thank you!

Out of scope:
Do any VS Code extensions do this?

**Comment by fwouts** [comment]
Hey @lgibso34, good questions.

There are no analytics in Preview.js (aside from the ones VS Code itself records) although I'm not excluding adding some for **Preview.js Pro** in the future (all Pro users must agree to the [privacy policy](https://previewjs.com/privacy)).

No, your source code isn't sent out, in fact the extension should work fine offline. There are no plans to ever do this.

All code, including Preview.js Pro, is of course public in this GitHub repo :)

In general, you can never know for sure that the code published on NPM or in the VS Code store is actually what you see on GitHub, so the absolute safest would probably be to clone the repo and build it yourself.

I haven't written instructions to do this, but in summary:
```
pnpm install
pnpm turbo run build --scope=@previewjs/core --include-dependencies
cd integrations/vscode
yarn install
yarn dev # installs the locally built version in Visual Studio Code
```

Edit: to answer your out of scope question, I'm sure some dodgy VS Code extensions do this. Microsoft approves updates of Preview.js within minutes, which makes me think the approval process is quite lightweight.

**Reply by lgibso34** [reply]
Thank you for your timely answer. Good luck!

**Reply by fwouts** [reply, marked as answer]
Starting with the next release expected in June or July 2024, Preview.js will start tracking how often specific features are used in order to better understand where to focus development resources. Please refer to https://previewjs.com/privacy for more information.

Your components source code will of course not be reported in analytics and will remain on the device.
